src/analytics/src/main/scala/osmesa/analytics/oneoffs/Roads.scala

```diff
@@ -65,11 +65,10 @@ object Analysis {
 
   /** How long is a Line, in metres? */
   private[this] def metres(line: Line): Double = {
-    val ps: Array[Point] = line.points // TODO Wasteful.
-    val head: Point = ps.head
-    val last: Point = ps.last
+    val ps: List[Point] = line.points.toList
+    val pairs: Iterator[(Point, Point)] = ps.iterator.zip(ps.tail.iterator)
 
-    Haversine(head.x, head.y, last.x, last.y)
+    pairs.foldLeft(0d) { case (acc, (p,c)) => acc + Haversine(p.x, p.y, c.x, c.y) }
```

Comment:
I had this as:

```scala
line.points.sliding(2).map(pair => Haversine(pair.head.x, pair.head.y, pair.last.x, pair.last.y)).reduce(_ + _)
```

Equivalent? (I'm coming up to speed on Scala, so advice helps.)

Reply:
I'll benchmark it.

Reply:
```
[info] Benchmark                Mode  Cnt   Score   Error  Units
[info] MetresBench.iterator10   avgt    5   2.732 ± 0.119  us/op
[info] MetresBench.iterator100  avgt    5  25.007 ± 0.393  us/op
[info] MetresBench.manual10     avgt    5   2.407 ± 0.044  us/op
[info] MetresBench.manual100    avgt    5  21.972 ± 0.317  us/op
[info] MetresBench.sliding10    avgt    5   4.863 ± 0.077  us/op
[info] MetresBench.sliding100   avgt    5  48.641 ± 0.420  us/op
[info] MetresBench.while10      avgt    5   2.070 ± 0.019  us/op
[info] MetresBench.while100     avgt    5  20.619 ± 0.076  us/op
```
`sliding` kills performance, looks like. My guess is from all the allocations of intermediate `Array`s that come in as the `pair` arg.